pyfifotax/data_structures_dataframe.py

```diff
@@ -0,0 +1,358 @@
+        return {k: None for k in tmp.keys()}
+
+    @staticmethod
+    def from_df_row(row: Series) -> DividendRow:
+        return DividendRow(
+            row.date, row.symbol, row.amount, row.tax_withholding, row.currency
+        )
+
+
+@dataclass
+class TaxWithholdingRow(DataFrameRow):
+    date: datetime
+    symbol: str
+    amount: pd.Float64Dtype
+    currency: str
+
+    @staticmethod
+    def from_schwab_json(json_dict: dict) -> TaxWithholdingRow:
+        date = datetime.strptime(json_dict["Date"], "%m/%d/%Y")
+        symbol = json_dict["Symbol"]
+        amount = pd.to_numeric(json_dict["Amount"].strip("-$").replace(",", ""))
+
+        return TaxWithholdingRow(
+            date,
+            symbol,
+            amount,
+            "USD",
+        )
+
+    @staticmethod
+    def empty_dict() -> dict:
+        tmp = TaxWithholdingRow(datetime(1, 1, 1), "", 0.0, "").to_dict()
+        return {k: None for k in tmp.keys()}
+
+
+@dataclass
+class SellOrderRow(DataFrameRow):
+    date: datetime
+    symbol: str
+    quantity: pd.Float64Dtype
+    sell_price: pd.Float64Dtype
+    fees: pd.Float64Dtype
+    currency: str
+
+    @staticmethod
+    def from_schwab_json(json_dict: dict) -> SellOrderRow:
+        date = datetime.strptime(json_dict["Date"], "%m/%d/%Y")
+        symbol = json_dict["Symbol"]
+        fees = pd.to_numeric(
+            json_dict["FeesAndCommissions"].strip("-$").replace(",", "")
+        )
+        quantity = pd.to_numeric(json_dict["Quantity"])
+        details = json_dict["TransactionDetails"]
+
+        sale_price = details[0]["Details"]["SalePrice"]
+        check_quantity = pd.to_numeric(0)
+        for det in details:
+            if not det["Details"]["SalePrice"] == sale_price:
+                raise ValueError(
+                    "Unexpected behavior when converting SellOrder data from Schwab JSON, check [TransactionDetails][i][Details][SalePrices] for inconsistent values"
+                )
+            check_quantity += pd.to_numeric(det["Details"]["Shares"])
+        if not check_quantity == quantity:
+            raise ValueError(
+                "Unexpected behavior when converting SellOrder data from Schwab JSON, check [TransactionDetails][i][Details][Shares] for inconsistent values"
+            )
+
+        sale_price = pd.to_numeric(sale_price.strip("$").replace(",", ""))
+
+        return SellOrderRow(
+            date,
+            symbol,
+            quantity,
+            sale_price,
+            fees,
+            "USD",
+        )
+
+    @staticmethod
+    def empty_dict() -> dict:
+        tmp = SellOrderRow(datetime(1, 1, 1), "", 0.0, 0.0, 0.0, "").to_dict()
+        return {k: None for k in tmp.keys()}
+
+    @staticmethod
+    def from_df_row(row: Series) -> SellOrderRow:
+        return SellOrderRow(
+            row.date, row.symbol, row.quantity, row.sell_price, row.fees, row.currency
+        )
+
+
+@dataclass
+class BuyOrderRow(DataFrameRow):
```

Comment:
I'll add again what I had in the previous version, i.e. a more general "comment"  column which then also can be used to indicate rows automatically imported by certain tools in case someone wants to mix-and-match manual and imported things or in the case one has to merge things outside of the 4-year history window of Schwab.